vars/updatecli.groovy

```diff
@@ -40,19 +43,56 @@ def call(userConfig = [:]) {
         org.jenkinsci.plugins.pipeline.modeldefinition.Utils.markStageSkippedForConditional(updatecliRunStage)
       }
     }
+
+    // If a custom version is provided, download and install that version of updatecli using tar files in a single sh step
+    if (runUpdatecli && finalConfig.version) {
+      stage("Download updatecli version ${finalConfig.version}") {
+        withEnv(["UPDATECLI_VERSION=${finalConfig.version}", "PATH+UPDATECLI=${customUpdatecliPath}"]) { //Extend PATH dynamically
+          sh '''
+            versionTag="v${UPDATECLI_VERSION}"
+            cpu="$(uname -m)"
+            # Construct the tar file name based on the CPU architecture.
+            tarFileName="updatecli_Linux_${cpu}.tar.gz" 
+            downloadUrl="https://github.com/updatecli/updatecli/releases/download/${versionTag}/${tarFileName}"
+            echo "Downloading updatecli version ${UPDATECLI_VERSION} from ${downloadUrl}"
+            curl --silent --location --output ${tarFileName} ${downloadUrl} || { echo "Download failed"; exit 1; }
+            # Create destination directory for extraction.
+            mkdir -p ${customUpdatecliPath}
+            # Extract the updatecli binary from the tar file.
+            tar --extract --gzip --file="${tarFileName}" --directory=${customUpdatecliPath} updatecli
+            # Remove the tar file leaving only the executable binary.
+            rm -f ${tarFileName}
+            # Debugging: Verify PATH and the resolved updatecli binary
+            echo "Updated PATH: $PATH"
+            echo "Resolved updatecli path: $(which updatecli)"
+          '''
+        }
+      }
+    }
+
+    // **Factorized updatecli command builder - defined once, after installation**
+    def updatecliCommand = finalConfig.version ? "${customUpdatecliPath}/updatecli" : "updatecli"
```

Comment:
this will not work for the non custom version usage ...

Reply:
as discussed with @jayfranco999 , in the current state of the code, it would. But a better solution is to rely on the `PATH` instead so this ternary form will go away, and the command will keep using `updatecli` as binary